### PR TITLE
Request to merge Feature/update-delete into dev branch

### PR DESCRIPTION
**Pull request by flemton**
In this PR the following commits were made to answer questions from the [exercise: query and update animals table](https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/vet_clinic_query_update_tables.md#vet-clinic-database-query-and-update-animals-table)

- [Add species to animals table](https://github.com/flemton/vet-clinic/commit/1f4cfd47b85761e1cc7c13955a6f6c0c5021539d)
Screenshot 2023-06-16 at 1 33 21 AM

Screenshot 2023-06-16 at 2 10 10 AM

- [Add transactions](https://github.com/flemton/vet-clinic/commit/0df45970703498692609f810ab894238de9e83b2)
Screenshot 2023-06-16 at 1 49 15 AM
Screenshot 2023-06-16 at 2 10 10 AM

- [Use aggregate functions](https://github.com/flemton/vet-clinic/commit/fce847f2e2afe2e14f4967cdeef53bc6f0673a7d)
Screenshot 2023-06-16 at 7 53 05 PM
Screenshot 2023-06-16 at 7 53 50 PM
Screenshot 2023-06-16 at 7 54 39 PM
Screenshot 2023-06-16 at 7 58 20 PM
Screenshot 2023-06-16 at 8 00 40 PM
Screenshot 2023-06-16 at 8 01 33 PM
Screenshot 2023-06-16 at 8 02 02 PM
Screenshot 2023-06-16 at 8 02 31 PM
Screenshot 2023-06-16 at 8 09 07 PM
Screenshot 2023-06-16 at 8 09 22 PM
Screenshot 2023-06-16 at 1 49 15 AM

### CODE REVIEW

File: schema.sql [hunk 1]
```diff
@@ -7,5 +7,6 @@ CREATE TABLE animals (
     escape_attempts INT,
     neutered        BOOLEAN,
     weight_kg       DECIMAL,
+    species         VARCHAR(250),
     PRIMARY KEY(id)
 );
```
Comment (1GORDON) [inline schema.sql:11]: Great job @flemton 🤝

- [ ] As stated in the project requirements, kindly add a new column called species with a string type. Ensure this command is added to the `schema.sql` file as well.
   `ALTER TABLE animals ADD species varchar(255);`

File: queries.sql [hunk 2]
```diff
@@ -8,3 +8,35 @@ SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
 SELECT * FROM animals WHERE neutered = true;
 SELECT * FROM animals WHERE NOT name = 'Gabumon';
 SELECT * FROM animals WHERE weight_kg  BETWEEN 10.4 and 17.3;
+
```
Comment (1GORDON) [inline queries.sql:11]: Great job @flemton 🤝

- [x] As stated in the project requirements, Inside a transaction update the `animals` table by setting the species column to unspecified. Verify that the change was made. Then roll back the change and verify that the species columns went back to the state before the transaction. Kindly add this transaction to the `queries.sql` file
- [x] Also remember to add a screenshot of this transaction to your pull request.

File: queries.sql [hunk 3]
```diff
@@ -8,3 +8,35 @@ SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
...
+BEGIN;
+SAVEPOINT begin;
+UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon';
+SAVEPOINT middle;
+UPDATE animals SET species = 'pokemon' WHERE name IS NULL;
+UPDATE animals SET species = 'pokemon' WHERE species IS NULL;
+SELECT * FROM animals;
+COMMIT;
+
```
Comment (1GORDON) [inline queries.sql:28]: Great job @flemton 🤝

- [x] As stated in the project requirements, Inside a transaction delete all records in the `animals` table, then roll back the transaction. Kindly add this transaction to the `queries.sql` file.

File: queries.sql [hunk 4]
```diff
@@ -8,3 +8,35 @@ SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
...
+
+BEGIN;
+DELETE FROM animals WHERE date_of_birth > '2022-01-01';
+SAVEPOINT delete;
+UPDATE animals SET weight_kg = weight_kg * -1;
+ROLLBACK TO SAVEPOINT delete;
+UPDATE animals SET weight_kg = weight_kg * -1 WHERE weight_kg < 0;
+COMMIT;
```
Comment (1GORDON) [inline queries.sql:41]: Great job @flemton 🤝

- [x] As stated in the project requirements, kindly add the results of this transaction to your pull request.

**Review by 1GORDON — CHANGES_REQUESTED** [review]
##
## Hi @flemton ✋

Good job so far!

## Highlights!
- Great use of Gitflow ✔️ 
- Good commit messages ✔️
- Descriptive Pull request ✔️

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️

- [ ] Kindly fix the broken issues link to make your readme file more professional.
- [ ] Kindly ensure all the screenshots for all queries are added to the pull request before you resubmit for review. You seem to have missed a number of screenshots. Kindly take a closer look at the [project requirements](https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/vet_clinic_query_update_tables.md).

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend considering them as they can improve your code._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@1GORDON` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: queries.sql (hunk 2 above)
Comment (flemton) [reply queries.sql:11]: Dear @1GORDON, please this transaction was done and it is the first screenshot under the 'Add transactions' list in the PR description.

File: queries.sql (hunk 3 above)
Comment (flemton) [reply queries.sql:28]: Dear @1GORDON, please this transaction was done and it is the first screenshot under the 'Add transactions' list in the PR description.

File: queries.sql (hunk 2 above)
Comment (flemton) [reply queries.sql:11]: Hi @1GORDON, this is done and it is the first screenshot of the PR description now, thanks.

File: queries.sql (hunk 4 above)
Comment (flemton) [reply queries.sql:41]: Dear @1GORDON, this is done and is now the last screenshot

File: queries.sql [hunk 5]
```diff
@@ -8,3 +8,41 @@ SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
 SELECT * FROM animals WHERE neutered = true;
 SELECT * FROM animals WHERE NOT name = 'Gabumon';
 SELECT * FROM animals WHERE weight_kg  BETWEEN 10.4 and 17.3;
+
+BEGIN;
+SAVEPOINT save1;
+DELETE FROM animals;
+ROLLBACK TO SAVEPOINT save1;
+COMMIT;
+
```
Comment (1GORDON) [inline queries.sql:19]: Great job @flemton 🤝

- [ ] As stated in the project requirements, Inside a transaction update the `animals` table by setting the species column to unspecified. Verify that the change was made. Then roll back the change and verify that the species columns went back to the state before the transaction. Kindly add this transaction to the `queries.sql` file.

**Review by 1GORDON — CHANGES_REQUESTED** [review]
##
## Hi @flemton ✋

Good job so far!

## Highlights!
- Followed Gitflow ✔️
- Professional Readme file ✔️

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend considering them as they can improve your code._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@1GORDON` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Reem-lab — APPROVED** [review]
# Hi @flemton,  👋🏻👋🏻

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights
- [ ] Great job on the funtionality 💯
- [ ] Followed Gitflow 🔥
- [ ] No Linters error ✅
- [ ] Kudos wokring on the requested changes⭐


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**